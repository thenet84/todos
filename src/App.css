body {
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

input[type=text] {
    flex: 10;
    padding: 5px;
}

form {
    display: flex;
    margin-bottom: 10px;
}

.container {
    margin: 0 auto;
    width: 70%;
}

.button {
    flex: 1;
    display: inline-block;
    border: none;
    background: #555;
    color: #fff;
    padding: 7px 20px;
    cursor: pointer;
}

.button:hover {
    background: #666;
}

.todoItemWrapper {
    background: #f4f4f4;
    padding: 2px;
    border-bottom: 1px #ccc dotted;
}

.todoItem {
    text-decoration: none;
}

.todoItemCompleted {
    text-decoration: line-through;
}

.deleteButton {
    background: #f00;
    color: #fff;
    border: none;
    border-radius: 25%;
    cursor: pointer;
    float: right;
    margin-right: 10px;
}

.deleteButton:hover {
    background: #803d3d;
}
