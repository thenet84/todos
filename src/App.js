import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import './App.css';

import { Header } from './components/layout/Header';
import { About } from './components/pages/About';
import { Home } from './components/pages/Home';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="container">
          <Header />
          <Route path="/" exact component={Home} />
          <Route path="/about" component={About} />
        </div>
      </Router>
    );
  }
}

export default App;
