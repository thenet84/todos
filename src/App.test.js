import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should add a new todo to the todo list', () => {
    const wrapper = shallow(<App />);

    wrapper
      .find('TodoForm')
      .props()
      .onChange({ preventDefault: jest.fn(), target: { value: 'hello' } });

    wrapper
      .find('TodoForm')
      .props()
      .onSubmit({ preventDefault: jest.fn(), target: { value: 'hello' } });

    expect(wrapper.find('TodoList').prop('todos')).toEqual([{ name: 'hello' }]);
  });
});
