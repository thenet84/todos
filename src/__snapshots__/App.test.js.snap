// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
<div>
  <TodoList
    todos={Array []}
  />
  <TodoForm
    onChange={[Function]}
    onSubmit={[Function]}
    term=""
  />
</div>
`;
